java/vector/src/main/java/org/apache/arrow/vector/BinaryPointerVector.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+import static org.apache.arrow.vector.NullCheckingForGet.NULL_CHECKING_ENABLED;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.ReferenceManager;
+import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+
+import io.netty.buffer.ArrowBuf;
+import io.netty.util.internal.PlatformDependent;
+
+/**
+ * Vector that contains pointers to a set of binary data.
+ */
+public class BinaryPointerVector extends FixedSizeBinaryVector {
+
+  /**
+   * The option for decoding a {@link BinaryPointerVector} in the
+   * receiver side of an IPC communication.
+   */
+  public enum DecodeType {
+    /**
+     * Decode as a {@link BinaryPointerVector}.
+     */
+    BINARY_POINTER,
+
+    /**
+     * Decode as a {@link VarBinaryVector}.
+     */
+    VAR_BINARY,
+
+    /**
+     * Decode as a {@link VarCharVector}.
+     */
+    VAR_CHAR
+  }
+
+  /**
```

Comment:
cant this be moved own and expressed as ADDRESS_WIDTH + LENGTH_WIDTH?

Reply:
Sure. Good suggestion.